agent.go

```diff
@@ -0,0 +1,40 @@
+// A simple proxy that multiplexes a unix socket connection
+//
+// Copyright 2017 HyperHQ Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package main
+
+import (
+	"syscall"
+
+	"github.com/hyperhq/runv/hyperstart/libhyperstart"
+)
+
+// A new shim agent implementation should have the shimAgent interface
+// Currently it is a subset of libhyperstart.Hyperstart interface but it
+// allows for new agent and protocols.
+type shimAgent interface {
+	SignalProcess(container, process string, signal syscall.Signal) error
+	WaitProcess(container, process string) int
+
+	WriteStdin(container, process string, data []byte) (int, error)
+	ReadStdout(container, process string, data []byte) (int, error)
+	ReadStderr(container, process string, data []byte) (int, error)
+	CloseStdin(container, process string) error
+
+	TtyWinResize(container, process string, row, col uint16) error
+}
+
+// libhyperstart.Hyperstart based shim agent
+func NewHyperAgent(addr string) (shimAgent, error) {
+	if h, err := libhyperstart.NewGrpcBasedHyperstart(addr); err != nil {
+		return nil, err
+	} else {
```

Comment:
nit: No need for the `else` here.

Reply:
It is _needed_ as `h` variable is defined in the `if` statement.

Reply:
@bergwolf Well not really, you can simplify like this:
```go
h, err := libhyperstart.NewGrpcBasedHyperstart(addr);
if err != nil {
		return nil, err
}

return h, nil
```

Reply:
@sboeuf It's just another version. I wouldn't call it simpler though :)